app/src/main/java/com/sun/makeupwindow/data/ProductDataSource.kt

```diff
@@ -0,0 +1,15 @@
+package com.sun.makeupwindow.data.source
+
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.utils.OnDataLoadCallback
+
+interface ProductDataSource {
+    interface Local{
+        fun getProduct(callback: OnDataLoadCallback<List<Product>>)
+        fun getItemProduct(idproduct: Int, callback: OnDataLoadCallback<Product>)
+        fun getProductByCategory(category: String,callback: OnDataLoadCallback<List<Product>>)
```

Comment:
```suggestion
        fun getProductsByCategory(category: String, callback: OnDataLoadCallback<List<Product>>)
```
tương tự với ProductDao, ColorDao